pkg/types/builder.go

```diff
@@ -39,28 +39,38 @@ const (
 // NewBuilder returns a new Builder.
 func NewBuilder(pkg *types.Package) *Builder {
 	return &Builder{
-		Package:  pkg,
-		comments: twtypes.Comments{},
+		Package: pkg,
+		Generated: Generated{
+			Comments: twtypes.Comments{},
+		},
 	}
 }
 
+// Generated stores the information of the generated types
+type Generated struct {
```

Comment:
I don't have a great suggestion but `Generated`/`types.Generated` feels too generic.

`types.TypeTree`
```suggestion
// TypeTree contains all the types and their sibling relations.
type TypeTree struct {
```

`types.CustomResourceDefinition`
```suggestion
// CustomResourceDefinition includes all Go types necessary to print a CustomResourceDefinition Go struct.
type CustomResourceDefinition struct {
```

`types.Managed`
```suggestion
// Managed includes all Go types necessary to print the Go structs for a Managed resource.
type Managed struct {
```

Reply:
I cannot claim that `types.Generated` is great but I couldn't find a better alternative and honestly would prefer it to the alternatives you suggested above :)

I believe it is not that bad and actually conveys its purpose: `types.Generated` is a struct that holds _Generated_ _types_